core/misc/autosubmit.js

```diff
@@ -37,12 +37,39 @@
 Backdrop.behaviors.autosubmit = {
   attach: function(context) {
     // 'this' references the form element
-    function triggerSubmit (e) {
+    function triggerSubmit (element) {
       var $this = $(this);
+      var $triggering_element = $(element);
+
+      // Variable "element" will have a value only when text fields trigger
+      // this. If element is undefined, then remove the data to prevent
+      // potential focus on a previously processed element.
+      if (element === undefined)  {
+        $('body').removeData('needs-focus');
+      }
+      else if ($triggering_element.parents('.view').length) {
+        var $triggering_element = $(element);
+        $('body').data('needs-focus', $triggering_element.attr('id'));
+      }
+
+      // Submit the form.
       $this.find('.autosubmit-click').trigger('click');
     }
 
-    // the change event bubbles so we only need to bind it to the outer form
+    // Listener to ajaxStop will re-focus on the text field as needed.
+    $(document).off('ajaxStop').on('ajaxStop', function () {
```

Comment:
This `.off('ajaxStop')` would unbind _any_ `ajaxStop` handler. This should be namespaced with `autosubmit` to only remove/add the handler from this file.